### PR TITLE
feat/link-math-validation

### PR DESCRIPTION
**Pull request by CarlaCostea**
integrate math-validation engine

### CODE REVIEW

File: packages/math-inline/configure/src/index.js [hunk 1]
```diff
@@ -13,10 +14,20 @@ import defaults from './defaults';
 const log = debug('pie-elements:math-inline:configure');
 
 export default class MathInlineConfigure extends HTMLElement {
-  static createDefaultModel = (model = {}) => ({
-    ...defaults.model,
-    ...model
-  });
+  static createDefaultModel = (model = {}) => {
+
+    // making sure that validation type is set
```
Comment (andreeapescar) [inline packages/math-inline/configure/src/index.js:19]: Update comment.

File: packages/math-inline/configure/src/response.jsx [hunk 2]
```diff
@@ -130,7 +132,23 @@ class Response extends React.Component {
     onResponseChange(newResponse, index);
   };
 
-  onAnswerChange = answer => {
+  onAllowTrailingZerosChange = (event) => {
+    const { response, onResponseChange, index } = this.props;
+    const newResponse = { ...response };
+
+    newResponse.allowTrailingZeros = !response.allowTrailingZeros;
```
Comment (andreeapescar) [inline packages/math-inline/configure/src/response.jsx:139]: Maybe we can create a custom function that gets a key and we'll have something like 
newResponse[key]= !response[key];

File: packages/math-inline/configure/src/response.jsx [hunk 3]
```diff
@@ -264,13 +293,40 @@ class Response extends React.Component {
...
           </div>
+          {validation === 'literal' && (
+            <div className={classes.flexContainer}>
+              {cAllowTrailingZeros.enabled && (
+                <FormControlLabel
+                  label={cAllowTrailingZeros.label}
+                  control={
+                    <Checkbox
+                      checked={allowTrailingZeros}
+                      onChange={this.onAllowTrailingZerosChange}
```
Comment (andreeapescar) [inline packages/math-inline/configure/src/response.jsx:311]: So here you'll have something like:
e => this.onCheckboxChange(e, 'allowTrailingZeros')

File: packages/math-inline/controller/src/index.js [hunk 4]
```diff
@@ -87,110 +43,47 @@ const getResponseCorrectness = (model, answerItem, isOutcome) => {
...
-function handleStringBasedCheck(acceptedValues, answerItem) {
-  let answerValueToUse = processAnswerItem(answerItem, true);
+function getIsAnswerCorrect(correctResponseItem, answerItem) {
   let answerCorrect = false;
 
-  for (let i = 0; i < acceptedValues.length; i++) {
-    let acceptedValueToUse = processAnswerItem(acceptedValues[i], true);
-
-    answerCorrect = answerValueToUse === acceptedValueToUse;
+  correctResponseItem.forEach(correctResponse => {
```
Comment (andreeapescar) [inline packages/math-inline/controller/src/index.js:50]: Sanity check

File: packages/math-inline/controller/src/index.js [hunk 5]
```diff
@@ -87,110 +43,47 @@ const getResponseCorrectness = (model, answerItem, isOutcome) => {
...
     if (!answerCorrect) {
       const acceptedValues = [correctResponse.answer].concat(
         Object.keys(correctResponse.alternates || {}).map(
           alternateId => correctResponse.alternates[alternateId]
         )
       );
 
-      if (correctResponse.validation === 'literal') {
+      try {
         for (let i = 0; i < acceptedValues.length; i++) {
```
Comment (andreeapescar) [inline packages/math-inline/controller/src/index.js:71]: Sanity check

File: packages/math-inline/controller/src/index.js [hunk 6]
```diff
@@ -237,18 +130,32 @@ export const outcome = (question, session, env) => {
...
-  feedbackEnabled: true,
-  promptEnabled: true,
-  rationaleEnabled: true,
-  teacherInstructionsEnabled: true,
-  studentInstructionsEnabled: true,
-  ...question
-});
+export const normalize = (question) => {
+
+  // making sure that validation type is set
```
Comment (andreeapescar) [inline packages/math-inline/controller/src/index.js:135]: Update comment

File: packages/math-inline/package.json [hunk 7]
```diff
@@ -14,6 +14,7 @@
   "dependencies": {
     "@material-ui/core": "^3.9.2",
     "@pie-framework/mathquill": "^1.1.0",
+    "@pie-framework/math-validation": "^1.0.0",
```
Comment (andreeapescar) [inline packages/math-inline/package.json:17]: Shouldn't this be in controller? ("@pie-lib/math-evaluator": "2.1.6" )

File: packages/pie-models/src/pie/math-inline/index.ts [hunk 8]
```diff
@@ -42,126 +54,156 @@ interface MathInlineResponse {
...
+
+  /**
+   * Indicates if partial scoring is allowed.
+   * This property is not used yet.
+   */
+  partialScoring?: boolean;
+
+  /**
+   * Indicates if the order of expression elements in literal validation can be ignore - whilst the expression is still mathematically correct
+   * @default is false
```
Comment (andreeapescar) [inline packages/pie-models/src/pie/math-inline/index.ts:111]: Make sure to update this properly and specify it's default

**Review by andreeapescar — CHANGES_REQUESTED** [review]

File: packages/math-inline/package.json (hunk 7 above)
Comment (CarlaCostea) [reply packages/math-inline/package.json:17]: it's in controller now. works fine

File: packages/math-inline/configure/src/response.jsx (hunk 2 above)
Comment (CarlaCostea) [reply packages/math-inline/configure/src/response.jsx:139]: done

File: packages/math-inline/configure/src/response.jsx (hunk 3 above)
Comment (CarlaCostea) [reply packages/math-inline/configure/src/response.jsx:311]: done

File: packages/math-inline/controller/src/index.js (hunk 5 above)
Comment (CarlaCostea) [reply packages/math-inline/controller/src/index.js:71]: ok

**Comment by CarlaCostea** [comment]
I solved the requested changes. Please check again @andreeapescar

**Review by andreeapescar — APPROVED** [review]

**Pull request merged**